Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/PayloadIntegrityTestClient/PayloadIntegrityTest.cs

```diff
@@ -0,0 +1,148 @@
+                                   {
+                                     Seconds = 3600 * 24,
+                                   },
+                     MaxRetries           = 3,
+                     Priority             = 1,
+                     EngineType           = EngineType.Unified.ToString(),
+                     ApplicationVersion   = "1.0.0-700",
+                     ApplicationService   = "ServiceApps",
+                     ApplicationName      = "ArmoniK.EndToEndTests.Worker",
+                     ApplicationNamespace = "ArmoniK.EndToEndTests.Worker.Tests.PayloadIntegrityTestWorker.Services",
+                   };
+
+
+    resultHandler_ = new ResultHandler((exception,
+                                        id) =>
+                                       {
+                                         Assert.Fail($"taskId:[{id}] failed with exception : {exception}");
+                                       },
+                                       (response,
+                                        id) =>
+                                       {
+                                         responseAndData_.TryAdd(id,
+                                                                 response.ToString() ?? string.Empty);
+                                       });
+  }
+
+  private          TaskOptions?                         taskOptions_;
+  private          ResultHandler?                       resultHandler_;
+  private readonly ConcurrentDictionary<string, string> taskAndData_     = new();
+  private readonly ConcurrentDictionary<string, string> responseAndData_ = new();
+  private          IConfigurationRoot                   _configuration;
+
+  [TestCase(1,
+            1,
+            1)]
+  [TestCase(1,
+            50,
+            1)]
+  [TestCase(5,
+            50,
+            5)]
+  public void CopyPayload(int maxConcurrentBuffers,
+                          int maxTasksPerBuffer,
+                          int maxParallelChannels)
+  {
+    var numberOfPayload = 100;
+
+    var fixture = new Fixture();
+    var tasks   = new List<Task>();
+    var props = new Properties(_configuration,
+                               taskOptions_)
+                {
+                  MaxConcurrentBuffers = maxConcurrentBuffers,
+                  MaxTasksPerBuffer    = maxTasksPerBuffer,
+                  MaxParallelChannels  = maxParallelChannels,
+                };
+    var service = ServiceFactory.CreateService(props);
+    for (var i = 0; i < numberOfPayload; i++)
+    {
+      tasks.Add(NewMethod(fixture,
+                          service));
+    }
+
+    Task.WaitAll(tasks.ToArray());
+    while (responseAndData_.Count < numberOfPayload)
+    {
+      Thread.Sleep(100);
+    }
+
+    CollectionAssert.AreEquivalent(taskAndData_,
+                                   responseAndData_);
+
+    service.Dispose();
+    responseAndData_.Clear();
+    taskAndData_.Clear();
+  }
+
+  private async Task NewMethod(Fixture fixture,
```

Comment:
NewMethod ?

Reply:
arf quick customer test integration. Renaming in the next push